src/securityPolicy.c

```diff
@@ -273,16 +273,22 @@ security_policy_t policyForSignTxInput()
 security_policy_t policyForSignTxOutputAddressBytes(
         sign_tx_signingmode_t txSigningMode,
         const uint8_t* rawAddressBuffer, size_t rawAddressSize,
-        const uint8_t networkId, const uint32_t protocolMagic
+        const uint8_t networkId, const uint32_t protocolMagic,
+        bool includeDatumHash
 )
 {
+
 	ASSERT(rawAddressSize < BUFFER_SIZE_PARANOIA);
 
 	// address type and network identification
 	ASSERT(rawAddressSize >= 1);
 	const address_type_t addressType = getAddressType(rawAddressBuffer[0]);
 	const uint8_t addressNetworkId = getNetworkId(rawAddressBuffer[0]);
 
+	if (includeDatumHash) {
+		DENY_UNLESS(determineSpendingChoice(addressType) == SPENDING_SCRIPT_HASH || determineStakingChoide(addressType) == STAKING_SCRIPT_HASH);
```

Comment:
Perhaps would be more readable if the boolean condition got a name.
```suggestion
		const bool isScriptAddress = 
				determineSpendingChoice(addressType) == SPENDING_SCRIPT_HASH ||
				determineStakingChoide(addressType) == STAKING_SCRIPT_HASH;
		DENY_UNLESS(isScriptAddress);
```

Reply:
Also below. Perhaps worth to make isScriptAddress a function.